### ISSUE TITLE
Render all XML doc comments in the editor as they appear in quickinfo

### ISSUE THREAD

**Issue by jmarolf** [dotnet/roslyn#39991]
When I click inside the XML doc comments show me the regular XML format. When I am just scrolling through the file, render the comment like it would appear in quickinfo.

**Comment by jmarolf** [comment]
cc: @nguerrera @jasonmalinowski @ryzngard

**Comment by sharwell** [comment]
Another option would be continuing to work on the replacement. 😄

**Comment by HaloFour** [comment]
I'd love the editor to collapse down doc comments into something very unintrusive but configurable.  Personally I'd want to see as little as the summary inline.  Hovering would display the full rendered quick info, and clicking some icon would display the entire comment for editing.

IMO, the format isn't particularly relevant.  I have to work with a lot of Java, Scala and Python as well and despite having "simpler" doc comment formats they're still a massive wall of text.

**Comment by Joe4evr** [comment]
> I'd love the editor to collapse down doc comments into something very unintrusive but configurable. Personally I'd want to see as little as the summary inline.

Well, at the very least, in VS, you can already click the `[-]` in the margin to collapse doc comments pretty much exactly like that. Also, I would *L O V E* to have an option to have VS show doc comments collapsed like that as the default.

**Comment by tannergooding** [comment]
@sharwell, could you elaborate on "replacement"?

**Comment by sharwell** [comment]
@tannergooding My work in progress is https://gist.github.com/sharwell/ab7a6ccab745c7e0a5b8662104e79735

**Comment by jmarolf** [comment]
> @sharwell  Another option would be continuing to work on the replacement. 😄

Sure, but this could be done in a sprint. rev'ing the whole documentation comment infrastructure is a much bigger proposition

**Comment by ryzngard** [comment]
I also see this as an opportunity to enrich markdown comment structures by showing what will be rendered in quick info inline with the comment.

**Comment by tannergooding** [comment]
I commented on https://github.com/dotnet/csharplang/issues/2987. I like this idea but think it would also be good to have some kind of language service API or IDE feature that allows a better editing experience as well.

That is, support for rendering documentation comments in a better format would be great and it would also be great if you could edit your documentation comments in a better format (e.g. Markdown, Doxygen, etc) but still have it saved to the recognized XML format on the backend (so the compiler and existing tooling doesn't need to update, etc).

**Comment by jmarolf** [comment]
yeah, maybe a better way for me to say it is that we can do both. I don't see a reason for one feature to wait for the other.

**Comment by sharwell** [comment]
One option I thought of: we could have a command that collapses the current comment to just one line (like you normally get when you collapse it from the outlining margin), and opens up a Peek window below it containing the comment text as Markdown instead of XML. When the Peek window is closed, the contents are converted back to XML to place in the documentation comment.
